### PR TITLE
net-irc/ngircd: add new OpenRC service

### PR DESCRIPTION
**Pull request by CyberTailor**
Ebuild changes (r3):
* update EAPI 7 -> 8
* uncomment PIDFile, ServerUID and ServerGID
* correct ngircd.conf's owner

Service changes (r2):
* use pidfile
* use declarative config
* add configtest command
* add reload command

### CODE REVIEW

File: net-irc/ngircd/files/ngircd.init-r2.d [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+pidfile="${NGIRCD_PIDFILE:-/var/run/ngircd/ngircd.pid}"
+
+depend() {
+	need net
+	use logger
+	provide ircd
+}
+
+start_pre() {
+	checkpath -f "${pidfile}" -o ngircd:ngircd || return 1
```
Comment (thesamesam) [inline net-irc/ngircd/files/ngircd.init-r2.d:27]:
```suggestion
	checkpath -f "${pidfile}" -o ${command_user} || return 1
```

File: net-irc/ngircd/files/ngircd.init-r2.d [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+stop_pre() {
+	if [ "${RC_CMD}" = "restart" ]; then
+		configtest || return 1
+	fi
+}
+
+reload() {
+	configtest || return 1
+	ebegin "Refreshing ngircd's configuration"
```
Comment (thesamesam) [inline net-irc/ngircd/files/ngircd.init-r2.d:41]:
```suggestion
	ebegin "Refreshing ${RV_SVCNAME}'s configuration"
```

File: net-irc/ngircd/files/ngircd.init-r2.d [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	start-stop-daemon --signal SIGHUP --pidfile "${pidfile}"
+	eend $? "Failed to reload ngircd"
```
Comment (thesamesam) [inline net-irc/ngircd/files/ngircd.init-r2.d:43]:
```suggestion
	eend $? "Failed to reload ${RV_SVCNAME}"
```

File: net-irc/ngircd/files/ngircd.init-r2.d [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+configtest() {
+	ebegin "Checking ngircd's configuration"
```
Comment (thesamesam) [inline net-irc/ngircd/files/ngircd.init-r2.d:47]:
```suggestion
	ebegin "Checking ${RV_SVCNAME}'s configuration"
```

(The reason for this is to more easily facilitate e.g. ngircd.foo instances).

File: net-irc/ngircd/files/ngircd.init-r2.d [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+description_configtest="Run ngircd's internal config check."
+description_reload="Reload the ngircd's configuration."
+
+: ${NGIRCD_CONFIGFILE:=/etc/ngircd/ngircd.conf}
+
+command="/usr/sbin/ngircd"
+command_args="${NGIRCD_OPTS} -f \"${NGIRCD_CONFIGFILE}\""
+command_args_foreground="-n"
+command_user="ngircd:ngircd"
+pidfile="${NGIRCD_PIDFILE:-/var/run/ngircd/ngircd.pid}"
```
Comment (thesamesam) [inline net-irc/ngircd/files/ngircd.init-r2.d:18]:
```suggestion
pidfile="${NGIRCD_PIDFILE:-/var/run/ngircd/${RV_SVCNAME}.pid}"
```

File: net-irc/ngircd/files/ngircd.init-r2.d [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+extra_commands="configtest"
+extra_started_commands="reload"
+
+description="IRC server"
+description_configtest="Run ngircd's internal config check."
+description_reload="Reload the ngircd's configuration."
+
+: ${NGIRCD_CONFIGFILE:=/etc/ngircd/ngircd.conf}
```
Comment (thesamesam) [inline net-irc/ngircd/files/ngircd.init-r2.d:12]: Maybe:
```suggestion
: ${NGIRCD_CONFIGFILE:=/etc/ngircd/${RV_SVCNAME}.conf}
```

but I suppose less important given we have it in .conf.

File: net-irc/ngircd/files/ngircd.init-r2.d (hunk 5 above)
Comment (CyberTailor) [reply net-irc/ngircd/files/ngircd.init-r2.d:18]: pidfile is set in ngircd.conf, so it needs to be explicitly changed by the user anyways

Comment (thesamesam) [reply net-irc/ngircd/files/ngircd.init-r2.d:18]: We can still pick a useful default.

Comment (CyberTailor) [reply net-irc/ngircd/files/ngircd.init-r2.d]: okay, done

**Pull request closed without merging**